Sources/SmithyJSON/Writer/Writer+JSONSerialization.swift

```diff
@@ -0,0 +1,56 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import struct Foundation.Data
+import class Foundation.JSONSerialization
+import class Foundation.NSNull
+import class Foundation.NSNumber
+
+extension Writer {
```

Comment:
These methods serialize data held in a tree of `Writer`s to JSON.
(Nodes with nil JSON are trimmed away before serializing; this "detaches" writers that ended up having nothing written to them without requiring the reading closures to explicitly call a `detach()` method.)